catalog/core/catalog-core-camelcomponent/src/main/java/ddf/camel/component/catalog/content/ContentProducerDataAccessObject.java

```diff
@@ -155,7 +164,9 @@ public void createContentItem(FileSystemPersistenceProvider fileIdMap, ContentEn
                 List<Metacard> createdMetacards = createResponse.getCreatedMetacards();
 
                 for (Metacard metacard : createdMetacards) {
-                    fileIdMap.store(key, metacard.getId());
+                    if (safeKey != null) {
```

Comment:
That was deliberate to preserve the existing behavior. If we dislike the logging loops I can refactor them away?